src/tests/jsonLogicFixtures.js

```diff
@@ -0,0 +1,1098 @@
+        then: {
+          required: ['field_c'],
+        },
+        else: {
+          properties: {
+            field_c: false,
+          },
+        },
+      },
+    ],
+  },
+};
+
+export const schemaWithChecksAndThenValidationsOnThen = {
+  properties: {
+    field_a: {
+      type: 'number',
+    },
+    field_b: {
+      type: 'number',
+    },
+    field_c: {
+      type: 'number',
+    },
+  },
+  required: ['field_a', 'field_b'],
+  'x-jsf-logic': {
+    validations: {
+      c_must_be_large: {
+        errorMessage: 'Needs more numbers',
+        rule: {
+          '>': [{ var: 'field_c' }, 200],
+        },
+      },
+      require_c: {
+        rule: {
+          and: [
+            { '!==': [{ var: 'field_b' }, null] },
+            { '!==': [{ var: 'field_a' }, null] },
+            { '>': [{ var: 'field_a' }, { var: 'field_b' }] },
+          ],
+        },
+      },
+    },
+    allOf: [
+      {
+        if: {
+          validations: {
+            require_c: {
+              const: true,
+            },
+          },
+        },
+        then: {
+          required: ['field_c'],
+          properties: {
+            field_c: {
+              description: 'I am a description!',
+              'x-jsf-requiredValidations': ['c_must_be_large'],
+            },
+          },
+        },
+        else: {
+          properties: {
+            field_c: false,
+          },
+        },
+      },
+    ],
+  },
+};
+
+export const schemaWithComputedValueChecksInIf = {
+  properties: {
+    field_a: {
+      type: 'number',
+    },
+    field_b: {
+      type: 'number',
+    },
+    field_c: {
+      type: 'number',
+    },
+  },
+  required: ['field_a', 'field_b'],
+  'x-jsf-logic': {
+    computedValues: {
+      require_c: {
+        rule: {
+          and: [
+            { '!==': [{ var: 'field_b' }, null] },
+            { '!==': [{ var: 'field_a' }, null] },
+            { '>': [{ var: 'field_a' }, { var: 'field_b' }] },
+          ],
+        },
+      },
+    },
+    allOf: [
+      {
+        if: {
+          computedValues: {
+            require_c: {
+              const: true,
+            },
+          },
+        },
+        then: {
+          required: ['field_c'],
+        },
+        else: {
+          properties: {
+            field_c: false,
+          },
+        },
+      },
+    ],
+  },
+};
+
+export const schemaWithMultipleComputedValueChecks = {
+  properties: {
+    field_a: {
+      type: 'number',
+    },
+    field_b: {
+      type: 'number',
+    },
+    field_c: {
+      type: 'number',
+    },
+  },
+  required: ['field_a', 'field_b'],
+  'x-jsf-logic': {
+    validations: {
+      double_b: {
+        errorMessage: 'Must be two times B',
+        rule: {
+          '>': [{ var: 'field_c' }, { '*': [{ var: 'field_b' }, 2] }],
+        },
+      },
+    },
+    computedValues: {
+      a_times_two: {
+        rule: {
+          '*': [{ var: 'field_a' }, 2],
+        },
+      },
+      mod_by_five: {
```

Comment:
naming: I noticed multiple places where the naming doesn't make sense and I can foresee this happening in the future. Perhaps we'll need to document guidelines on how to name these rules. Let me give you some examples:

- Incomplete naming:
  - This one should be `b_mod_by_five`
  - At `schemaWithIfStatementWithComputedValuesAndValidationChecks`, the `greater_than_b` should be `a_greater_than_b` (or even `a_>_b`)
- Desbribe the action, not the outcome
- At `schemaWithGreaterThanChecksForThreeFields` the `require_c` should be `a_greater_than_b`. Then based on this validation, we apply wtv outcome it's needed (in that example, the c gets required)

There are other cases through the examples, but I think you go the point. Does this make sense to you? :)